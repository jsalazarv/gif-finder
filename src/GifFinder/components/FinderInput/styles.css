.finder {

}

.finder__input {
    width: 561px;
    height: 44px;
    font-size: 16px;
    border-radius: 50px;
    padding-inline-start: 86px;
    padding-inline-end: 44px;
    background-color: #1D1D1D;
    border: 3px solid #222222;
    color: #1C1C1C;
}

.wrapper {
    position: relative;
    display: inline-block;
    width: 418px;
}
.wrapper input {
    background-color: transparent;
    border: none;
    outline: none;
    width: 100%;
    font-size: 1em;
    box-sizing: border-box;
    padding-bottom: 5px;
    border-bottom: 2px solid #797979;
    text-align: center;
    color: #ffffff;
}
.wrapper input:focus + .underline {
    width: 100%;
}
.wrapper .underline {
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
    height: 2px;
    width: 0;
    background: linear-gradient(
        115deg,
        #4fcf70,
        #fad648,
        #a767e5,
        #12bcfe,
        #44ce7b
    );
    transition: 0.5s;
}
