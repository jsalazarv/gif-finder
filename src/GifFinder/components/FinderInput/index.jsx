import { useState } from 'react';
import './styles.css';

export const FinderInput = ({ onAddCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onAddCategories(inputValue.trim());
  };

  return (
    <div className="finder">
      <form role="form" onSubmit={onSubmit}>
        <div className="wrapper">
          <input
            type="text"
            placeholder="Hello, search all the GIFs..."
            value={inputValue}
            onChange={onInputChange}
          />
          <span className="underline"></span>
        </div>
      </form>
    </div>
  );
};
