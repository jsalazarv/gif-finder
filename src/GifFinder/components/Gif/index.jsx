import './styles.css';

export const Gif = ({ title, url }) => {
  return (
    <div className="gif">
      <div className="image-container">
        <img src={url} alt={title} />
      </div>
      <a role="anchor" href={url} target="_blank">
        <small>{title}</small>
      </a>
    </div>
  );
};
