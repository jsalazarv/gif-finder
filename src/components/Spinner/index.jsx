import './styles.css';

export const Spinner = () => {
  return (
    <div data-testid="spinner" className="fingerprint-spinner">
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
      <div className="spinner-ring"></div>
    </div>
  );
};
