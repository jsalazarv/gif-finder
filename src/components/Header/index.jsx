import { useState } from 'react';

import { clsx } from 'clsx';

import { FinderInput } from '../../GifFinder/components/FinderInput';
import './styles.css';

export const Header = ({ search }) => {
  const [isSearching, setIsSearching] = useState(false);

  const onSearchPerformed = (item) => {
    setIsSearching(true);
    search(item);
  };

  return (
    <header className={clsx('header', { collapsed: isSearching })}>
      <nav className="navbar__content">
        <h1 className="navbar__title">
          <span className="animate-text">GIF</span>
          INDER
        </h1>
        <FinderInput onAddCategories={onSearchPerformed} />
      </nav>
    </header>
  );
};
