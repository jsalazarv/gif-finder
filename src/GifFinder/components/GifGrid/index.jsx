import { useFetchGifs } from '../../hooks/useFetchGifs.js';
import { Spinner } from '../../../components/Spinner';
import { Gif } from '../Gif';
import './styles.css';

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  if (isLoading) {
    return (
      <div className="gif-grid h-screen flex flex-col justify-center items-center">
        <Spinner />
        <span className="text-red-500 text-xs">LOADING</span>
      </div>
    );
  }

  return (
    <div className="gif-grid">
      {gifs.map((gif) => (
        <Gif key={gif.id} {...gif} />
      ))}
    </div>
  );
};
