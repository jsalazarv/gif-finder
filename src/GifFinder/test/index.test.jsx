import { render } from '@testing-library/react';
import { GifFinder } from '../index';
import { FinderInput } from '../components/FinderInput';

jest.mock('../components/FinderInput');

describe('[GifFinder]', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('[Snapshot]', () => {
    test('Should render and contain class [finder-container]', () => {
      const { container } = render(<GifFinder />);

      expect(container.firstChild).toHaveClass('finder-container');
      expect(container).toMatchSnapshot();
    });

    test('Should render <FinderInput /> component', () => {
      render(<GifFinder />);

      expect(FinderInput).toHaveBeenCalledOnce();
    });
  });
});
