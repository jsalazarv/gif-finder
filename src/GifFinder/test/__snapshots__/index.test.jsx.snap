// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GifFinder] [Snapshot] Should render and contain class [finder-container] 1`] = `
<div>
  <div
    class="finder-container"
  />
</div>
`;
